### PR TITLE
feat: Added advanced typings and GlobalBrowserApi

### PR DESCRIPTION
**Pull request by StyleT**

### CODE REVIEW

File: doc/ilc_app_interface.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+[can be found here](https://namecheap.github.io/ilc-sdk/interfaces/app.applifecyclefnprops.html).
+
+
+
+### Init code during app bundle loading
+
+Sometimes you need to run some initialization code right after app bundle will be loaded in the browser and usually you
+want to be able to pass some configuration properties to that code.
+
+ILC allows you to export a function called `mainSpa(props)` that will receive application properties that were defined in
```
Comment (nightnei) [inline doc/ilc_app_interface.md:23]: should we add information about default export?
https://github.com/namecheap/ilc/blob/master/ilc/client/BundleLoader.js#L73

File: src/app/interfaces/CustomProps.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface CustomProps {
+    [str: string]: any;
+    [num: number]: any;
```
Comment (nightnei) [inline src/app/interfaces/CustomProps.ts:3]: why `number`?

**Review by nightnei — APPROVED** [review]

File: src/app/interfaces/CustomProps.ts (hunk 2 above)
Comment (StyleT) [reply src/app/interfaces/CustomProps.ts:3]: I assume to be able to have numeric keys. I took this interface from single-spa as is

Comment (nightnei) [reply src/app/interfaces/CustomProps.ts:3]: it's strange, but anyway it's not a problem, so let it be as is

**Pull request merged**